### PR TITLE
Device Auth Flow

### PR DESCRIPTION
**Pull request by ewgenius**

### CODE REVIEW

**Review by sgrebnov — APPROVED** [review]

File: bin/spice/pkg/api/spice_api_client.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+func (s *SpiceApiClient) GetAuthUrl(authCode string) string {
+	return fmt.Sprintf("%s/auth/device?code=%s", s.baseUrl, authCode)
+}
+
+func (s *SpiceApiClient) GetUser(accessToken string) (user SpiceUser, err error) {
+	var spiceUser SpiceUser
+
+	request, err := http.NewRequest("GET", fmt.Sprintf("%s/api/device/user", s.baseUrl), nil)
+	if err != nil {
+		fmt.Println(err)
```
Comment (phillipleblanc) [inline bin/spice/pkg/api/spice_api_client.go]: Need to return here.

File: bin/spice/pkg/cli/cmd/login.go [hunk 2]
```diff
@@ -25,25 +30,56 @@ spice login
...
+		var accessToken string
+
+		cmd.Println("Waiting for authentication...")
+		// poll for auth status
+		for {
+			time.Sleep(time.Second)
+
+			authStatusResponse, err := spiceApiClient.ExchangeCode(authCode)
+			if err != nil {
+				fmt.Println("Error:", err)
```
Comment (phillipleblanc) [inline bin/spice/pkg/cli/cmd/login.go]:
```suggestion
				cmd.Println("Error:", err)
```

File: bin/spice/pkg/api/spice_api_client.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	}
+
+	return nil
+}
+
+func (s *SpiceApiClient) GetAuthUrl(authCode string) string {
+	return fmt.Sprintf("%s/auth/device?code=%s", s.baseUrl, authCode)
+}
+
+func (s *SpiceApiClient) GetUser(accessToken string) (user SpiceUser, err error) {
```
Comment (phillipleblanc) [inline bin/spice/pkg/api/spice_api_client.go]:
```suggestion
func (s *SpiceApiClient) GetUser(accessToken string) (SpiceUser, error) {
```

File: bin/spice/pkg/cli/cmd/login.go [hunk 4]
```diff
@@ -25,25 +30,56 @@ spice login
...
+
+			if authStatusResponse.AccessToken != "" {
+				accessToken = authStatusResponse.AccessToken
+				break
+			}
+		}
+
+		user, err := spiceApiClient.GetUser(accessToken)
+		if err != nil {
+			fmt.Println("Error:", err)
```
Comment (phillipleblanc) [inline bin/spice/pkg/cli/cmd/login.go]:
```suggestion
			cmd.Println("Error:", err)
```

File: bin/spice/pkg/api/spice_api_client.go (hunk 1 above)
Comment (phillipleblanc) [inline bin/spice/pkg/api/spice_api_client.go]: Since we log the error in the caller - we don't need to log errors here and below.

Comment (ewgenius) [reply bin/spice/pkg/api/spice_api_client.go]: Removed logging

Comment (ewgenius) [reply bin/spice/pkg/api/spice_api_client.go]: Added return

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**